### PR TITLE
Adding in simple Feature Toggle for Datastore Selection

### PR DESCRIPTION
**Pull request by guypritchard**
A simple feature toggle to use the InMemory Datastore.
```json
"FeatureToggles" : {
  "InMemoryStore": true
}
```
or

```yaml
- FEATURETOGGLES__INMEMORYSTORE=true
```

To enable.

### CODE REVIEW

File: src/CentOps.Api/appsettings.Development.json [hunk 1]
```diff
@@ -13,5 +13,8 @@
   },
   "ConnectionStrings": {
     "AdminApiKey": "testingadmin"
+  },
+  "FeatureToggles": {
```
Comment (martinkearn) [inline src/CentOps.Api/appsettings.Development.json:17]: I wonder if it is worth adding some notes to the repo's readme to give an overview of what the setting does and give examples for both appsettings.json and things like docker compose files?

**Review by martinkearn — APPROVED** [review]

File: src/CentOps.Api/Configuration/FeatureToggles.cs [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿namespace CentOps.Api.Configuration
+{
+    public class FeatureToggles
+    {
+        public bool InMemoryStore { get; set; }
```
Comment (decodingahmed) [inline src/CentOps.Api/Configuration/FeatureToggles.cs]: Sorry for being pedantic but perhaps `UseInMemoryStore`?

File: src/CentOps.Api/Program.cs [hunk 3]
```diff
@@ -31,12 +30,19 @@ public static void Main(string[] args)
...
+            var features = builder.Configuration.GetSection("FeatureToggles").Get<FeatureToggles>();
+            if (features != null && features.InMemoryStore)
+            {
+                services.AddInMemoryDataStores();
+            }
+            else
+            {
+                var section = builder.Configuration.GetSection("CosmosDb");
+                services.AddCosmosDbServices(section);
+            }
```
Comment (decodingahmed) [inline src/CentOps.Api/Program.cs]: nit: Might be a bit neater to move it to extension an extension method in the extensions file anyway?

PS: slightly surprised that code analysis isn't complaining about too many namespaces and types in the same file 🤷‍♂️

**Review by decodingahmed — APPROVED** [review]
Looks good, just a couple of nit-picks - feel free to dismiss or defer them :)

File: src/CentOps.Api/Configuration/FeatureToggles.cs (hunk 2 above)
Comment (guypritchard) [reply src/CentOps.Api/Configuration/FeatureToggles.cs]: Yep - happy to change that.

File: src/CentOps.Api/Program.cs (hunk 3 above)
Comment (guypritchard) [reply src/CentOps.Api/Program.cs]: I don't think I brought in any namespaces but you're right it looks tidier - implemented.  Ta!

**Review by yahya130 — APPROVED** [review]

**Pull request merged**